CaImAn/caiman_parameters.py

```diff
@@ -0,0 +1,129 @@
+            self.preview_parameters = danse_parameters[defs.Parameters.Main.PREVIEW]
+
+        with h5py.File(self.paths[defs.Parameters.Path.FILEPATH], 'r') as file_:
+            self.dataname  = defs.DoricFile.Dataset.IMAGE_STACK if defs.DoricFile.Dataset.IMAGE_STACK in file_[self.paths[defs.Parameters.Path.H5PATH]] else defs.DoricFile.Deprecated.Dataset.IMAGES_STACK
+
+            images = np.array(file_[f"{self.paths[defs.Parameters.Path.H5PATH]}/{self.dataname}"])
+
+            freq    = utils.get_frequency(file_, f"{self.paths[defs.Parameters.Path.H5PATH]}/{defs.DoricFile.Dataset.TIME}")
+            dims, T = utils.get_dims(file_, f"{self.paths[defs.Parameters.Path.H5PATH]}/{self.dataname}")
+
+        neuron_diameter = tuple([self.parameters[defs.Parameters.danse.NEURO_DIAM_MIN], self.parameters[defs.Parameters.danse.NEURO_DIAM_MAX]])
+
+        self.params_caiman = {
+            "fr": freq,
+            "dims": dims,
+            "decay_time": 0.4,
+            "pw_rigid": True,
+            "max_shifts": (neuron_diameter[0], neuron_diameter[0]),
+            "gSig_filt": (neuron_diameter[0], neuron_diameter[0]),
+            "strides": (neuron_diameter[-1]*4, neuron_diameter[-1]*4),
+            "overlaps": (neuron_diameter[-1]*2, neuron_diameter[-1]*2),
+            "max_deviation_rigid": neuron_diameter[0]/2,
+            "border_nan": "copy",
+            "method_init": "corr_pnr",  # use this for 1 photon
+            "K": None,
+            "gSig": (neuron_diameter[0], neuron_diameter[0]),
+            "merge_thr": 0.8,
+            "p": 1,
+            "tsub": self.parameters[defs.Parameters.danse.TEMPORAL_DOWNSAMPLE],
+            "ssub": self.parameters[defs.Parameters.danse.SPATIAL_DOWNSAMPLE],
+            "rf": neuron_diameter[-1]*4,
+            "stride": neuron_diameter[-1]*2,
+            "only_init": True,    # set it to True to run CNMF-E
+            "nb": 0,
+            "nb_patch": 0,
+            "method_deconvolution": "oasis",       # could use "cvxpy" alternatively
+            "low_rank_background": None,
+            "update_background_components": True,  # sometimes setting to False improve the results
+            "min_corr": self.parameters[defs.Parameters.danse.LOCAL_CORR_THRESHOLD],
+            "min_pnr": self.parameters[defs.Parameters.danse.PNR_THRESHOLD],
+            "normalize_init": False,               # just leave as is
+            "center_psf": True,                    # leave as is for 1 photon
+            "ssub_B": 2,
+            "ring_size_factor": 1.4,
+            "del_duplicates": True,
+            "use_cnn": False,
+            "fnames": self.paths[defs.Parameters.Path.TMP_DIR] + '/' + f"tiff_{'_'.join(self.get_h5path_names()[2:4])}.tif"
+            }
+
+        print(cm_defs.Messages.WRITE_IMAGE_TIFF, flush=True)
+        imwrite(self.params_caiman["fnames"], images.transpose(2, 0, 1))
+        del images
+
+        self.opts = params.CNMFParams(params_dict = self.params_caiman)
```

Comment:
what `opts` stands for? would `cnmf_params` be beter?

Reply:
I think it's for options. `cnmf_params` would be indeed better! 

For the moment I just did a copy past of most of the variable and make stuff works. I think it could be nice to change some of them if you see variables with strange names please let me know